src/inspector_socket.cc

```diff
@@ -48,16 +49,17 @@ static void dump_hex(const char* buf, size_t len) {
 }
 #endif
 
+static void remove_from_beginning(std::vector<char>* buffer, size_t count) {
```

Comment:
I used a pointer here to indicate that the parameter will be modified... Should I use a non-const ref?